components/core/src/clp_s/SchemaMap.cpp

```diff
@@ -4,7 +4,7 @@
 #include "ZstdCompressor.hpp"
 
 namespace clp_s {
-int32_t SchemaMap::add_schema(std::set<int32_t>& schema) {
+int32_t SchemaMap::add_schema(Schema const& schema) {
```

Comment:
Even though this function is not a part of the PR, but I would suggest to rewrite it in the following way:
```
int32_t SchemaMap::add_schema(Schema const& schema) {
    auto const schema_it = m_schema_map.find(schema);
    if (m_schema_map.end() == schema_it) {
        return schema_it->second;
    }
    m_schema_map.emplace(schema, m_current_schema_id);
    return m_current_schema_id++;
}
```
Reasons:
1. const correctness on the iterator
2. Yoda condition
3. Trying to avoid inner return of an if-else statement (as clang-tidy would also alert)
4. Avoid using `[]` to insert element to a map/unordered_map

Reply:
Makes sense to me, I'll change it.

Reply:
oh yeah, sorry for the mistake :D